openpype/pipeline/colorspace.py

```diff
@@ -132,52 +197,87 @@ def get_imageio_colorspace_from_filepath(
     return colorspace_name
 
 
+def get_colorspace_from_filepath(config_path, filepath):
+    """Get colorspace from file path wrapper.
+
+    Wrapper function for getting colorspace from file path
+    with use of OCIO v2 file-rules.
+
+    Args:
+        config_path (str): path leading to config.ocio file
+        filepath (str): path leading to a file
+
+    Returns:
+        Any[str, None]: matching colorspace name
+    """
+    if not compatibility_check():
+        # python environment is not compatible with PyOpenColorIO
+        # needs to be run in subprocess
+        result_data = get_wrapped_with_subprocess(
+            "colorspace", "get_colorspace_from_filepath",
+            config_path=config_path,
+            filepath=filepath
+        )
+        if result_data:
+            return result_data[0]
+
+    from openpype.scripts.ocio_wrapper import _get_colorspace_from_filepath
+
+    result_data = _get_colorspace_from_filepath(config_path, filepath)
+
+    if result_data:
+        return result_data[0]
+
+
 def parse_colorspace_from_filepath(
-    path, host_name, project_name,
-    config_data=None,
-    project_settings=None
+    filepath, colorspaces=None, config_path=None
 ):
     """Parse colorspace name from filepath
 
     An input path can have colorspace name used as part of name
     or as folder name.
 
+    Example:
+        >>> config_path = "path/to/config.ocio"
+        >>> colorspaces = get_ocio_config_colorspaces(config_path)
+        >>> colorspace = parse_colorspace_from_filepath(
+                "path/to/file/acescg/file.exr",
+                colorspaces=colorspaces
+            )
+        >>> print(colorspace)
+        acescg
+
     Args:
-        path (str): path string
-        host_name (str): host name
-        project_name (str): project name
-        config_data (dict, optional): config path and template in dict.
-                                      Defaults to None.
-        project_settings (dict, optional): project settings. Defaults to None.
+        filepath (str): path string
+        colorspaces (Optional[dict[str]]): list of colorspaces
+        config_path (Optional[str]): path to config.ocio file
 
     Returns:
         str: name of colorspace
     """
-    if not config_data:
-        project_settings = project_settings or get_project_settings(
-            project_name
+    if not colorspaces and not config_path:
+        raise ValueError(
+            "You need to provide `config_path` if you don't "
+            "want to provide input `colorspaces`."
```

Comment:
```suggestion
        raise ValueError(
            "Missing either required input `colorspaces` or `config_path`."
```
_If you don't want_ doesn't really sound like what I'd expect in a developer message.

Or:
```suggestion
        raise ValueError(
            "Must provide at least one of `colorspaces` or `config_path` input arguments."
```